boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dtsi

```diff
@@ -64,6 +64,19 @@
 		};
 	};
 
+	nrf5340_audio_dk: nrf5340-audio-dk {
+		compatible = "nrf5340_audio_dk";
+		hw-codec-in-gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+		hw-codec-irq-in-gpios = <&gpio0 19 GPIO_ACTIVE_LOW>;
+		hw-codec-reset-out-gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+		hw-codec-sel-out-gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+		spi-sel-in-gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+		pmic-iset-out-gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
+		board-id-en-out-gpios =  <&gpio0 24 GPIO_ACTIVE_HIGH>;
+		board-id-in-gpios = <&gpio0 27 GPIO_ACTIVE_HIGH>;
+		curr-mon-alert-in-gpios =  <&gpio1 15 GPIO_ACTIVE_LOW>;
+	};
```

Comment:
-1 on that, this is not correct usage of DT. Please create proper compatibles for each IP block, and potentially drivers where needed.

Reply:
I guess it is not possible. I suggest using [/zephyr,user](https://docs.zephyrproject.org/latest/build/dts/zephyr-user-node.html#the-zephyr-user-node) and move this changes (last push) to downstream [nrf noup]?

Reply:
Probably derived from [downstream board support](https://github.com/nrfconnect/sdk-nrf/blob/6c606e116b8ef0f015a9e256ee557118f2d33f7b/boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts#L95-L98) and not detected during review.

Reply:
Shouldn't be hard to create bindings and then set up proper dts nodes for these?

Reply:
I found out that the solutiona bove does not work anyway. I get some involved dts errors.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/55675 will require rework in downstream code, the way those props were defined/used was wrong. Happy to help with that if needed.

Reply:
@koffes Please remove https://github.com/zephyrproject-rtos/zephyr/compare/b26002ef0e972c957294f050c281f3710beb41d7..ece78dbe1cb3a21ab5c9df7d1b32d8bcebdfcb72